[Location] Location 컴포넌트 및 LocationPinIcon 컴포넌트, LocationContainer 컴포넌트 초안

dani-day opened the pull request:
## 🔍️ 이 PR을 통해 해결하려는 문제가 무엇인가요?

> 어떤 기능을 구현한건지, 이슈 대응이라면 어떤 이슈인지 PR이 열리게 된 계기와 목적을 Reviewer 들이 쉽게 이해할 수 있도록 적어 주세요
> 일감 백로그 링크나 다이어그램, 피그마를 첨부해도 좋아요

- 오시는 길 페이지의 주요 레이아웃, 스타일링, 기능 구현

https://github.com/ReOpenSG/ReOpenSG/assets/134567470/d989bbdb-9a91-4ebe-b3f2-d46e11317994


## ✨ 이 PR에서 핵심적으로 변경된 사항은 무엇일까요?

> 문제를 해결하면서 주요하게 변경된 사항들을 적어 주세요

- 오시는 길 페이지의 주요 레이아웃, 스타일링, 기능 구현을 완료하였습니다.
- CSS 모듈화 완료하였습니다.
- 반응형 구현 완료하였습니다.
- 컴포넌트화 완료하였습니다.

## 🔖 핵심 변경 사항 외에 추가적으로 변경된 부분이 있나요?

> 없으면 "없음" 이라고 기재해 주세요

- 없음

## 🙏 Reviewer 분들이 이런 부분을 신경써서 봐 주시면 좋겠어요

> 개발 과정에서 다른 분들의 의견은 어떠한지 궁금했거나 크로스 체크가 필요하다고 느껴진 코드가 있다면 남겨주세요

- 기능 구현, 또는 반응형 이슈 중심으로 리뷰해주시면 감사하겠습니다 :)

## 🩺 이 PR에서 테스트 혹은 검증이 필요한 부분이 있을까요?

> 테스트가 필요한 항목이나 테스트 코드가 추가되었다면 함께 적어주세요

- 없음

### 📌 PR 진행 시 이러한 점들을 참고해 주세요

- Reviewer 분들은 코드 리뷰 시 좋은 코드의 방향을 제시하되, 코드 수정을 강제하지 말아 주세요.
- Reviewer 분들은 좋은 코드를 발견한 경우, 칭찬과 격려를 아끼지 말아 주세요.
- Review는 특수한 케이스가 아니면 Reviewer로 지정된 시점 기준으로 3일 이내에 진행해 주세요.
- Comment 작성 시 Prefix로 P1, P2, P3 를 적어 주시면 Assignee가 보다 명확하게 Comment에 대해 대응할 수 있어요
  - P1 : 꼭 반영해 주세요 (Request Changes) - 이슈가 발생하거나 취약점이 발견되는 케이스 등
  - P2 : 반영을 적극적으로 고려해 주시면 좋을 것 같아요 (Comment)
  - P3 : 이런 방법도 있을 것 같아요~ 등의 사소한 의견입니다 (Chore)

#

---

## 📝 Assignee를 위한 CheckList

- [ ] To-Do Item

dani-day committed:
✨ [Feat] : Location 컴포넌트 마크업 및 스타일링

+ Location 페이지의 마크업 및 스타일링을 완료하였습니다.
+ 세계지도 이미지 및 핀 아이콘 반응형으로 구현하였습니다.

! 핀 아이콘 hover 시 tooltip 은 반응형 구현이 필요합니다.
! 구글 링크 아이콘 및 링크 기능 추가가 필요합니다.

Related to #61

dani-day committed:
✨ [Feat] : 지도 새창 링크 기능 및 아이콘 추가

+ 주소 별 새창 링크를 연결하였습니다.

Related to: #61

dani-day committed:
🌈 [Design] : 반응형 구현 CSS 코드 수정

+ 좀 더 부드럽게 움직이도록 반응형 구현 코드 수정하였습니다.

Related to: #61

dani-day committed:
🐛 [Fix] : 모바일, 태블릿 크기에서의 지도 핀 위치 조정

+ 모바일, 태블릿 크기에서 지도 핀 위치 조정이 필요하여 수정하였습니다.

Related to: #61

ingbinsee commented on src/components/Location/LocationContainer.jsx:
```diff
@@ -0,0 +1,54 @@
+          <div className={styles.locationContact}>
+            <p>
+              Tel. &nbsp;
+              {locationTel}
```
P3: <LInk />에 tel:00-000-0000 속성을 추가하여 클릭 시 전화로 연결될 수 있으면 좋겠습니다.

dani-day replied:
수정하였습니다 :) 클릭 시 전화 연결 여부 확인이 어려워보여 전화번호 hover 시 underline 되는 스타일링 추가하였습니다.

ingbinsee commented on src/views/Location.jsx:
```diff
@@ -1,7 +1,33 @@
+      </div>
+      <section className={styles.locationSection}>
+        <div className={styles.locationWrapper}>
+          <LocationContainer locationName="본사" locationAddress="우 18469 경기도 화성시 동탄첨단산업1로 51-9 엠타워 1201호" locationTel="031-373-2612" locationFax="031-377-2615" locationImage={mTowerImage} locationLink="https://maps.app.goo.gl/Lda5XHqKMYWHyQpYA" />
```
P3: 우 18469 뒤에 ')' 추가 요청드립니다. 또한, 시안과 같이 우편번호와 주소 간 줄바꿈이 있으면 좋을 것 같습니다.

![image](https://github.com/ReOpenSG/ReOpenSG/assets/140426866/93268fb1-a554-45a7-a5eb-9531d78ab556)

<img width="259" alt="image" src="https://github.com/ReOpenSG/ReOpenSG/assets/140426866/52f4f21b-1f24-42d2-ac83-d004b4312df7">

dani-day replied:
수정하였습니다 :)

ingbinsee reviewed (approved):
아래 파일에 의견 남겨두었습니다.

- LocationContainer.jsx(P3 review)
- Location.jsx(P3 review)

kelvin-chihyun reviewed (approved):
은빈님이 요청해주신 부분만 조치해주시면 될 것 같습니다!

ryudg reviewed (approved):
특이사항 없습니다!!!

dani-day committed:
🐛 [Fix] : 1차 PR 리뷰사항 반영

+ Tel 에 링크 태그 추가하여 전화 연결되도록 하였습니다. hover 시 style 추가하였습니다.
+ 우편번호 레이아웃 수정하였습니다.

Related to: #78
